system-parachains/asset-hubs/asset-hub-polkadot/src/ah_migration/call_filter.rs

```diff
@@ -0,0 +1,118 @@
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Call filters for Asset Hub during the Asset Hub Migration.
+
+use crate::*;
+
+/// Contains all calls that are enabled during the migration.
+pub struct CallsEnabledDuringMigration;
+impl Contains<<Runtime as frame_system::Config>::RuntimeCall> for CallsEnabledDuringMigration {
+	fn contains(call: &<Runtime as frame_system::Config>::RuntimeCall) -> bool {
+		let (during, _after) = call_allowed_status(call);
+		if !during {
+			log::warn!("Call bounced by the filter during the migration: {:?}", call);
+		}
+		during
+	}
+}
+
+/// Contains all calls that are enabled after the migration.
+pub struct CallsEnabledAfterMigration;
+impl Contains<<Runtime as frame_system::Config>::RuntimeCall> for CallsEnabledAfterMigration {
+	fn contains(call: &<Runtime as frame_system::Config>::RuntimeCall) -> bool {
+		let (_during, after) = call_allowed_status(call);
+		if !after {
+			log::warn!("Call bounced by the filter after the migration: {:?}", call);
+		}
+		after
+	}
+}
+
+/// Return whether a call should be enabled during and/or after the migration.
+///
+/// Time line of the migration looks like this:
+///
+/// --------|-----------|--------->
+///       Start        End
+///
+/// We now define 2 periods:
+///
+/// 1. During the migration: [Start, End]
+/// 2. After the migration: (End, ∞)
+///
+/// Visually:
+///
+/// ```text
+///         |-----1-----|
+///                      |---2---->
+/// --------|-----------|--------->
+///       Start        End
+/// ```
+///
+/// This call returns a 2-tuple to indicate whether a call is enabled during these periods.
+pub fn call_allowed_status(call: &<Runtime as frame_system::Config>::RuntimeCall) -> (bool, bool) {
+	use RuntimeCall::*;
+	const ON: bool = true;
+	const OFF: bool = false;
+
+	let during_migration = match call {
+		AhOps(..) => OFF,
+		AhMigrator(..) => ON,
+		AssetConversion(..) => OFF,
+		AssetRate(..) => OFF,
+		Assets(..) => OFF,
+		Balances(..) => OFF,
+		Bounties(..) => OFF,
+		ChildBounties(..) => OFF,
+		Claims(..) => OFF,
+		CollatorSelection(..) => OFF, // TODO maybe disable them since staking is also disabled?
+		ConvictionVoting(..) => OFF,
+		CumulusXcm(..) => OFF, /* Empty call enum, see https://github.com/paritytech/polkadot-sdk/issues/8222 */
+		FastUnstake(..) => OFF,
+		ForeignAssets(..) => OFF,
+		Indices(..) => OFF,
+		MessageQueue(..) => ON, // TODO think about this
+		Multisig(..) => OFF,
+		NominationPools(..) => OFF,
+		Nfts(..) => OFF,
+		ParachainInfo(..) => OFF, /* Empty call enum, see https://github.com/paritytech/polkadot-sdk/issues/8222 */
+		ParachainSystem(..) => ON, // Only inherent and root calls
+		PolkadotXcm(..) => OFF,
+		PoolAssets(..) => OFF,
+		Preimage(..) => OFF,
+		Proxy(..) => OFF,
+		Referenda(..) => OFF,
+		Scheduler(..) => OFF,
```

Comment:
all calls are permissioned. we can keep it on for emergency (cancel some scheduled call). but minor, since I do not see real use case.